### DISCUSSION TITLE
Are there performance benefits using React Query over Redux/Redux-persist?

### DISCUSSION THREAD

**Discussion by IHIutch** [TanStack/query#2462] [Q&A]
I'm new to React Query and though I understand the different use cases between Query and Redux, obviously Redux is often used to manage server state, which is something I'm used to doing. As I'm getting more familiar with Query, I'm wondering if there are actual performance gains from using it instead of Redux/Redux-persist. Does the browser cache have benefits over localStorage (Is there a significant difference at all?)? Or is the value of Query just in simplifying managing server state? (I know Query is packed with all sorts of cool features as well, but generally speaking)

Thanks 😊

**Comment by TkDodo** [comment]
> Does the browser cache have benefits over localStorage

I don't think that is comparable. The react-query cache lives in the memory of the browser, so if you reload the page, it's gone. Persisting to localstorage is a completely different topic, which react-query currently supports via the experimental webstorage persistor: https://react-query.tanstack.com/plugins/persistQueryClient

**Reply by IHIutch** [reply]
Gotcha. So there's not a performance benefit from using one or the other, necessarily? In regards to reading the stored/cached data, that is.

**Reply by TkDodo** [reply]
both are pretty optimized

**Reply by IHIutch** [reply]
Thanks, that's helpful. My technical knowledge of the browser clients themselves is rather limited
